src/bits/shared.h

```diff
@@ -53,6 +53,12 @@ namespace detail {
     template <typename FN, typename... ARGS>
     using invoke_result_t = typename invoke_result<FN, ARGS...>::type;
 
+    // Test if two variables is at the same address.
+    template <typename T, typename S>
+    bool is_same_object(const T &t, const S &s)
+    {
+        return static_cast<const void *>(&t) == static_cast<const void *>(&s);
```

Comment:
Nag, you're meant to use std::addressof for this. Probably not worth making it "truly general", I guess.

(Scenario I'm talking about: `t` is a sub-object of `s` (e.g. Base and Derived). A comparison between `T*` and `S*` may still work, with the compiler doing the necessary pointer adjustments. However if you compare the pointers as `void*`, the comparison it will fail even if the two pointers do point to the same object.)